src/components/CandidateForm.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import styles from './CandidateForm.module.css'
+
+const CandidateForm = (props) => {
+    let defaultState = Object.assign({}, ...props.inputData.map(input => {
+        const {value, name} = input;
+        return {
+            [name]: value
+        }
+    }));
+    const [state, setState] = useState(defaultState);
+
+    const handleOnChange = (e) => {
+        e.preventDefault();
+        const {name, value} = e.target;
+        setState(prevState => {
+            return {
+                ...prevState,
+                [name]: value
+            }
+        });
+    };
+
+    const formInputs = props.inputData.map(input => {
+        const {name, placeholder} = input;
+        return (    
+            <input
+                key={name}
+                type="text"
+                name={name}
+                value={`${state[name]}`}
+                placeholder={placeholder}
+                onChange={(e) => handleOnChange(e)}
+                />
+        )   
+    });
+    
+    return (
+        <form className={`${styles.form} ${styles.field}`}>
+            {formInputs}
+            <input type="submit" value="Submit" onClick={(e) => props.handleSubmitButton(e, state)}/>
```

Comment:
Instead of sending the e evnet here, you could create a local function called handleSubmit(), that would e.preventDefault, and then call the parent function props.handleSubmitButton().

The form component should know how to manage itself, Account or Register page should not know anything or change any behaviour of the CandidateForm component. This is done so we do not create tightly coupled components.